### PR TITLE
Improve UI

### PR DESCRIPTION
**Pull request by gdozot2**
There are two things in this PR:

* First, when a setView is applied with new view, all the listener should be updated. Do you think there is a better way or need a comment where this update is done?
* Do you think the time is correct for the warning message outside the border or do you think another solution?

### CODE REVIEW

File: src/components/mode/create.ts [hunk 1]
```diff
@@ -68,15 +82,14 @@ export default class SingleCreate {
...
-      if (zoom && resolution && zoom > minZoomAllowed ) {
-        vectorLayer.setStyle(function (feature) {          
-          return CreateStyle.setupCircles(feature, (zoom / resolution));
-        });
-      }
-    })
+    if (useStore().getOptions().border.url !== '') {
+      window.addEventListener('border-contraint-enabled', () => {
+        map.getView().un('change:resolution', () => this.setChangeResolution(map, vectorLayer))
+        map.getView().on('change:resolution', () => this.setChangeResolution(map, vectorLayer))
```
Comment (slecorne) [inline src/components/mode/create.ts]: why do we need that? a comment would be good£
and similar for others

File: src/styles/controls.css [hunk 2]
```diff
@@ -14,6 +14,11 @@
   flex-direction: column;
 }
 
+.ol-control button:focus {
+  background-color: var(--control-background-color-hover) !important;
+  outline: none !important;
```
Comment (slecorne) [inline src/styles/controls.css]: try avoiding !important and adding additional css selector instead

File: src/utils/control-icon-manager.ts [hunk 3]
```diff
@@ -56,16 +58,26 @@ export default class ControlIconManager {
...
-      });
+    if (options.enableRotation) {
+      if (options.border.url !== '') {
+        window.addEventListener('border-contraint-enabled', () => {
+          useStore().getMap().getView().un('change:rotation', (event) => setRotationChange(map, event, leftControlIconContainer.div))
+          useStore().getMap().getView().on('change:rotation', (event) => setRotationChange(map, event, leftControlIconContainer.div))
+        })
+      }
+      map.getView().on('change:rotation', (event) => setRotationChange(map, event, leftControlIconContainer.div));
+    }
```
Comment (slecorne) [inline src/utils/control-icon-manager.ts]: this code is duplicated in many places, should be probably refactor in an helper function like `registerMapEvent(event, callback)`

**Review by slecorne — CHANGES_REQUESTED** [review]

File: src/components/mode/create.ts (hunk 1 above)
Comment (gdozot2) [reply src/components/mode/create.ts]: Comment in the new EventManager

File: src/styles/controls.css (hunk 2 above)
Comment (gdozot2) [reply src/styles/controls.css]: Done

File: src/utils/control-icon-manager.ts (hunk 3 above)
Comment (gdozot2) [reply src/utils/control-icon-manager.ts]: Done

**Review by slecorne — APPROVED** [review]

**Pull request merged**